### ISSUE TITLE
Use sphinx to generate documentation from python files

### ISSUE THREAD

**Issue by LonelyCat124** [stfc/HartreeParticleDSL#9]
Some documentation can be created automatically from the pydoc documentation. This needs to be done inside any documentation.

**Comment by sreyan-ghosh** [comment]
I am not able to figure it out, I am facing problems with the source directory and the project structure. Need help.

**Comment by LonelyCat124** [comment]
What issues are you seeing? All of the project's code is in `src/HartreeParticleDSL` with the `test` folder containing the pytest suite.
Things outside of the src can definitely be adjusted if needs be - they were really just a minimal setup to get the project installed with pip and allow development.

**Comment by sreyan-ghosh** [comment]
So, I am not very familiar with sphinx but I looked at a couple of videos on youtube and read through the docs. The problem I am facing is, I am not able to figure out what to add in the `index.rst` and the `conf.py` files. I have tried pointing it to the `src/HartreeParticleDSL` directory and also to the root, and also did a `os.walk()` to individually add the dirs, but that too gave a similar error. I cross-referenced the error on Google and got the idea that it refers to an incorrect source dir pointing problem. I'll share snapshots of the error soon.
